### PR TITLE
Add a toolbar to the entire page layout

### PR DESCRIPTION
**Pull request by calebj0seph**
Did this as it's going to be needed to add action buttons (like an 'Add' button on the Overview page). Had to refactor the Page and NavigationMenu components quite a bit in the process.

I've designed this so that we can specify our own toolbar component for each route, meaning its completely customisable between pages.

I've based this off GH-13, can rebase it off development after GH-13 is merged.

### CODE REVIEW

**Comment by PatrickShaw** [comment]
@calebj0seph note that it's called Material Design **Guidelines**. We don't have follow it word for word. I feel like UX fundamentals (like wasted screen real-estate) are more important than making sure our app looks like every other material design app.

I'm fine with a toolbar+nav drawer but it's only useful to have both if we actually need both. 
Do you think we're going to make full use of both?

**Comment by calebj0seph** [comment]
We'll go with the Atlassian-style toolbar instead, we do need a space for a page title and a small set of actions.

**Comment by PatrickShaw** [comment]
We can still put it in. I'm just saying it's worth considering how much screen real-estate we need to allocate to navigation and high level actions cause I don't think anyone (including me) has actually thought about it yet.

**Comment by calebj0seph** [comment]
I think Atlassian looks nicer anyway, will need to do it as a separate PR though since the toolbar could just be included on each page directly.

File: src/view/Overview.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import React, { SFC } from 'react';
-import { SpecificationList } from 'basic/SpecificationList';
 import { observer } from 'mobx-react';
+import Typography from 'material-ui/Typography';
 import { state } from 'state/SpecificationState';
-import { ContentContainer } from 'basic/ContentContainer';
+import { ContentContainer } from 'view/basic/ContentContainer';
+import { SpecificationList } from 'view/basic/SpecificationList';
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply src/view/Overview.tsx]: That's what i thought too but again, I don't want to type `view/basic` several times. I was worried about the chance that `basic` ends up being another folder too but if that were to happen it wouldn't be particularly hard to replace all occurrences of `basic` with another alias.

Some simple regex would probably do. E.g. `from 'basic/([\w/]+)'` and replace it with `from 'xxx/$1'`

**Pull request closed without merging**